crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -180,22 +180,47 @@ impl JsonRpcHandler {
                 transaction_type = TransactionType::Declare;
                 Transaction::Declare(crate::api::models::transaction::DeclareTransaction::Version2(
                     DeclareTransactionV2 {
-                        class_hash: declare_v2.class_hash.unwrap(),
+                        class_hash: declare_v2.class_hash.unwrap_or_default(),
                         sender_address: ContractAddressHex(declare_v2.sender_address),
                         nonce: declare_v2.nonce,
                         max_fee: Fee(declare_v2.max_fee),
                         version: Felt::from(2),
-                        transaction_hash: declare_v2.transaction_hash.unwrap(),
+                        transaction_hash: declare_v2.transaction_hash.unwrap_or_default(),
                         signature: declare_v2.signature,
                         compiled_class_hash: declare_v2.compiled_class_hash,
                     },
                 ))
             }
-            starknet_core::transactions::Transaction::DeployAccount(_deploy) => {
-                return Err(error::ApiError::TransactionNotFound);
+            starknet_core::transactions::Transaction::DeployAccount(deploy) => {
+                transaction_type = TransactionType::DeployAccount;
+                Transaction::DeployAccount(
+                    crate::api::models::transaction::DeployAccountTransaction {
+                        transaction_hash: Felt::from(deploy.inner.hash_value().clone()),
+                        max_fee: Fee(deploy.max_fee),
+                        version: deploy.version,
+                        signature: deploy.signature.clone(),
+                        nonce: deploy.nonce,
+                        class_hash: deploy.class_hash().unwrap_or_default(),
```

Comment:
unwrap_or_default in this case operates on Result, in case of an error it will hide it behind a default value

Reply:
I removed mapping code and use `Ok(TransactionWithType::try_from(&transaction.inner)?)` in 3df0fb73f32aaf36ef00556c45495078384dcc96